client/src/components/Sharing/Sharing.vue

```diff
@@ -0,0 +1,469 @@
+                extra: defaultExtra(),
+            },
+            shareFields: ["email", { key: "id", label: "" }],
+            makeMembersPublic: false,
+            showUrl: true,
+            tooltipClipboard: "Copy URL",
+            shareWithEmail: "",
+            actions: {
+                enable_link_access: "enable_link_access",
+                disable_link_access: "disable_link_access",
+                publish: "publish",
+                unpublish: "unpublish",
+                share_with: "share_with",
+                unshare_with: "unshare_with",
+            },
+            share_option: {
+                make_public: "make_public",
+                make_accessible_to_shared: "make_accessible_to_shared",
+                no_changes: "no_changes",
+            },
+        };
+    },
+    computed: {
+        permissionsChangeRequired() {
+            if (!this.item.extra) return false;
+            return (
+                this.item.extra && (this.item.extra.can_change.length > 0 || this.item.extra.cannot_change.length > 0)
+            );
+        },
+        pluralNameLower() {
+            return this.plural_name.toLowerCase();
+        },
+        itemStatus() {
+            return this.item.published ? "accessible via link and published" : "accessible via link";
+        },
+        itemRoot() {
+            const port = window.location.port ? `:${window.location.port}` : "";
+            return `${window.location.protocol}//${window.location.hostname}${port}${getAppRoot()}`;
+        },
+        itemUrl() {
+            return `${this.itemRoot}${this.item.username_and_slug}`;
+        },
+        itemSlugParts() {
+            const str = this.item.username_and_slug;
+            const index = str.lastIndexOf("/");
+            return [str.substring(0, index + 1), str.substring(index + 1)];
+        },
+        itemUrlParts() {
+            const str = this.itemUrl;
+            const index = str.lastIndexOf("/");
+            return [str.substring(0, index + 1), str.substring(index + 1)];
+        },
+        published_url() {
+            return `${getAppRoot()}${this.pluralNameLower}/list_published`;
+        },
+        slugUrl() {
+            return `${getAppRoot()}api/${this.pluralNameLower}/${this.id}/slug`;
+        },
+    },
+    created: function () {
+        this.getModel();
+    },
+    methods: {
+        addError(newError) {
+            // temporary turning Set into Array, until we update till Vue 3.0, that supports Set reactivity
+            this.errors = Array.from(new Set(this.errors).add(newError));
+        },
+        onCopy() {
+            copy(this.itemUrl);
+            this.tooltipClipboard = "Copied!";
+        },
+        onCopyOut() {
+            this.tooltipClipboard = "Copy URL";
+        },
+        onEdit() {
+            this.showUrl = false;
+        },
+        assignItem(newItem) {
+            if (newItem.errors) this.errors = newItem.errors;
```

Comment:
Well, I don't want to show old errors from the previous request. Right before this call I do `this.errors = [];`

> Also see my comment below about oneline if statements.

Once we decide the linting faith in https://github.com/galaxyproject/galaxy/pull/11925, sure. Once, we get it merged prettier will do one style for the whole application. So I'll wait for a bot to do it :) . I hope you're fine with this.

Reply:
> Well, I don't want to show old errors from the previous request. Right before this call I do `this.errors = [];`
> 
> >

But that's confusing then, because you're not undoing the previous failed operation in the interface, or did I miss that ?

Reply:
> > Well, I don't want to show old errors from the previous request. Right before this call I do `this.errors = [];`
> > >
> 
> But that's confusing then, because you're not undoing the previous failed operation in the interface, or did I miss that ?

No worries, it's getting re-rendered here https://github.com/galaxyproject/galaxy/blob/a3516c8cfe3921916f32cdb30a027feb186ce713/client/src/components/Sharing/Sharing.vue#L5